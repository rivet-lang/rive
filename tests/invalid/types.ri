extern (C) {
    fn malloc(sz: usize) *void;
    fn free(pointer: &void) void; // FAIL
}

struct St {
    f1: &i32; // FAIL

    f2: (i32, i32, i32, i32, i32, i32, i32, i32, i32);

    f3: void;

    f4: pkg;
    f5: =;
}

fn main() { }
