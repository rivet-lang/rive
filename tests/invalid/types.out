tests/invalid/types.ri:9:9: error: tuples can have a maximum of 8 types
  9 |     f2: (i32, i32, i32, i32, i32, i32, i32, i32, i32);
    |         ^
    = help: you can use a struct instead
tests/invalid/types.ri:14:9: error: expected type, found token `=`
  14 |     f5: =;
     |         ^
rivetc: error: could not compile module `types`, aborting due to 2 previous errors